third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -2931,43 +2970,96 @@ pub mod debug {
  *   is to involve an explicit representation of the type layout.
  *
  **************************************************************************************/
+use crate::values::ValueExchange;
 use serde::{
     de::Error as DeError,
     ser::{Error as SerError, SerializeSeq, SerializeTuple},
     Deserialize,
 };
 
 impl Value {
+    /// Deserializes bytes into a Move value as is.
     pub fn simple_deserialize(blob: &[u8], layout: &MoveTypeLayout) -> Option<Value> {
-        bcs::from_bytes_seed(SeedWrapper { layout }, blob).ok()
+        bcs::from_bytes_seed(
+            SeedWrapper {
+                exchange: None,
+                layout,
+            },
+            blob,
+        )
+        .ok()
     }
 
+    /// Serializes a Move value into bytes as is.
     pub fn simple_serialize(&self, layout: &MoveTypeLayout) -> Option<Vec<u8>> {
         bcs::to_bytes(&AnnotatedValue {
+            exchange: None,
             layout,
             val: &self.0,
         })
         .ok()
     }
+
+    /// Deserializes bytes into a Move value, swapping marked values with
+    /// identifiers based an the `exchange`.
+    pub fn deserialize_with_exchange(
```

Comment:
Notes:
let's try to have a separate value_exchange.rs file that implements functions that transform values into values, e.g. u64 to a u64, and then this part of the implementation can be more generic and simplified.